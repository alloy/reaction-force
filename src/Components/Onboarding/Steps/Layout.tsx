import React from "react"
import styled from "styled-components"

import Button from "../../Buttons/Ghost"
import { media } from "../../Helpers"
import StyledTitle from "../../Title"

interface Props {
  title: string
  subtitle: string
  onNextButtonPressed?: () => void
}

const Container = styled.div`
  max-width: 930px;
  margin-left: auto;
  margin-right: auto;
  ${media.sm`
    margin: 20px;
  `};
`

const MainTitle = styled(StyledTitle)`
  text-align: center;
  ${media.sm`
    text-align: left;
  `};
`
const Subtitle = styled(StyledTitle)`
  color: #999;
  margin-bottom: 100px;
  text-align: center;
  ${media.sm`
    text-align: left;
    margin-bottom: 15px;
    font-size: 20px
  `};
`

const ButtonContainer = styled(Button)`
  margin: 0 auto 50px;
  display: block;
  width: 250px;
  ${media.sm`
    width: 100%;
  `};
`

export class Layout extends React.Component<Props, null> {
  render() {
    const disabled = !this.props.onNextButtonPressed
    return (
      <Container>
        <MainTitle titleSize="xlarge">{this.props.title} </MainTitle>
        <Subtitle titleSize="xlarge">{this.props.subtitle}</Subtitle>
        <div>{this.props.children}</div>
        <ButtonContainer
          disabled={disabled}
          onClick={this.props.onNextButtonPressed}
        >
          Next
        </ButtonContainer>
      </Container>
    )
  }
}
