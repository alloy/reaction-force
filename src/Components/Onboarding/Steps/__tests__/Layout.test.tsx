import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { Layout } from "../Layout"

describe("Layout", () => {
  describe("snapshot", () => {
    it("renders the main layout", () => {
      const component = renderer
        .create(
          <Layout
            onNextButtonPressed={jest.fn()}
            title="Layout Title"
            subtitle="Layout Subtitle"
          />
        )
        .toJSON()
      expect(component).toMatchSnapshot()
    })
  })
})
