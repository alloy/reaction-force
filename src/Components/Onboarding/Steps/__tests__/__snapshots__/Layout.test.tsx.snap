// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout snapshot renders the main layout 1`] = `
.c6 {
  background: #e5e5e5;
  color: black;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 15px 30px;
  font-size: 13px;
  line-height: 1;
  outline: 0;
  -webkit-transition: background-color .25s,color .25s;
  transition: background-color .25s,color .25s;
  margin: 10px;
  border: none;
  box-sizing: border-box;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c6:hover:not(:disabled) {
  cursor: pointer;
  background: #cccccc;
}

.c5 {
  background: white;
  color: black;
  border: 1px solid #e5e5e5;
}

.c5:hover:not(:disabled) {
  background: white;
  color: #6e1fff;
}

.c2 {
  font-size: 50px;
  color: inherit;
  margin: 20px 0;
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
}

.c0 {
  max-width: 930px;
  margin-left: auto;
  margin-right: auto;
}

.c1 {
  text-align: center;
  line-height: normal;
}

.c3 {
  color: #999999;
  margin-bottom: 100px;
  text-align: center;
  line-height: normal;
}

.c4 {
  margin: 0 auto 50px;
  display: block;
  width: 250px;
}

@media (max-width:48em) {
  .c2 {
    font-size: 25px;
  }
}

@media (max-width:48em) {
  .c0 {
    margin: 20px;
  }
}

@media (max-width:48em) {
  .c1 {
    text-align: left;
  }
}

@media (max-width:48em) {
  .c3 {
    text-align: left;
    margin-bottom: 15px;
    font-size: 20px;
  }
}

@media (max-width:48em) {
  .c4 {
    width: 100%;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    color="inherit"
  >
    Layout Title
     
  </div>
  <div
    className="c3 c2"
    color="inherit"
  >
    Layout Subtitle
  </div>
  <div />
  <button
    className="c4 c5 c6"
    disabled={false}
    onClick={[Function]}
  >
    <span>
      Next
    </span>
  </button>
</div>
`;
